### PR TITLE
improvement(add validation): Improve snapshot nemesis - add snapshot validation

### PR DESCRIPTION
**Pull request by juliayakovlev**
From 'listsnapshots' output we can validate that all asked entities are in the snapshot.
For example, if snapshot was run on keyspace, we can check that all tables/MVs are in the snapshot

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/utils/common.py [hunk 1]
```diff
@@ -1951,3 +1951,26 @@ def ec2_ami_get_root_device_name(image_id, region):
...
+        Snapshot name Keyspace name Column family name True size Size on disk
+        1599414845162 system_schema keyspaces          0 bytes   71.71 KB
+        1599414845162 system_schema scylla_tables      0 bytes   73.21 KB
+        1599414845162 system_schema tables             0 bytes   81.48 KB
+        1599414845162 system_schema columns            0 bytes   80.12 KB
+
+        Total TrueDiskSpaceUsed: 0 bytes
+    """
+    snapshot_content = []
+    for line in listsnapshots_output.stdout.splitlines():
```
Comment (bentsi) [inline sdcm/utils/common.py:1970]: expecting unneeded objects and calling them "output" is a bad SW dev practice
`listsnapshots_output.stdout` -> `listsnapshots_output`

File: sdcm/utils/common.py [hunk 2]
```diff
@@ -1951,3 +1951,26 @@ def ec2_ami_get_root_device_name(image_id, region):
             return image.root_device_name
     except (TypeError, ClientError):
         raise AssertionError(f"Image '{image_id}' details not found in '{region}'")
+
+
+def get_snapshot_content(snapshot_name, listsnapshots_output) -> list():
```
Comment (bentsi) [inline sdcm/utils/common.py:1956]: add type hints to the function arguments

Comment (bentsi) [reply sdcm/utils/common.py:1956]: `get_snapshot_content` -> `parse_nodetool_listsnapshots`

Comment (bentsi) [reply sdcm/utils/common.py:1956]: the function should return dict with the details for each snapshot:

```python
SnapshotDetails = namedtuple("SnapshotDetails", ["keyspace_name", "table_name"])
...
{
    "1599414845162": SnapshotDetails(keyspace_name='system_schema', table_name='keyspaces'),
}

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -1474,6 +1474,25 @@ def target_node_reported_un_by_others():
         assert self.tester.hints_sending_in_progress() is False, "Hints are sent too slow"
         self.tester.verify_no_drops_and_errors(starting_from=start_time)
 
+    def _validate_snapshot(self, nodetool_cmd, snapshot_content):
```
Comment (bentsi) [inline sdcm/nemesis.py:1477]: 1) add type hints
2) add docstring that explains how this method validates snapshot

**Review by bentsi — CHANGES_REQUESTED** [review]

File: sdcm/utils/common.py (hunk 1 above)
Comment (juliayakovlev) [reply sdcm/utils/common.py:1970]: > expecting unneeded objects and calling them "output" is a bad SW dev practice
> `listsnapshots_output.stdout` -> `listsnapshots_output`

I don't understand. Maybe do you mean:
```listsnapshots_output.stdout -> listsnapshots.stdout```

Comment (juliayakovlev) [reply sdcm/utils/common.py]: Fixed

File: sdcm/utils/common.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/utils/common.py]: Done

File: sdcm/nemesis.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/nemesis.py]: Done

File: sdcm/utils/common.py [hunk 4]
```diff
@@ -1940,3 +1940,25 @@ def ec2_ami_get_root_device_name(image_id, region):
             return image.root_device_name
     except (TypeError, ClientError):
         raise AssertionError(f"Image '{image_id}' details not found in '{region}'")
+
+
+def parse_nodetool_listsnapshots(listsnapshots_output: str) -> defaultdict():
```
Comment (bentsi) [inline sdcm/utils/common.py:1945]: remove () from `-> defaultdict()`

Comment (juliayakovlev) [reply sdcm/utils/common.py]: fixed

**Review by bentsi — APPROVED** [review]

**Pull request merged**